src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -979,7 +986,25 @@ fn preprocess_link(
         }
     };
 
-    if should_ignore_link(path_str) {
+    // If there's no backticks, be lenient revert to old behavior.
+    // This is to prevent churn by linting on stuff that isn't meant to be a link.
+    // only shortcut links have simple enough syntax that they
+    // are likely to be written accidentlly, collapsed and reference links
+    // need 4 metachars, and reference links will not usually use
+    // backticks in the reference name.
+    // therefore, only shortcut syntax gets the lenient behavior.
+    //
+    // here's a truth table for how link kinds that cannot be urls are handled:
+    //
+    // |-------------------------------------------------------|
+    // |              |  is shortcut link  | not shortcut link |
+    // |--------------|--------------------|-------------------|
+    // | has backtick |    never ignore    |    never ignore   |
+    // | no backtick  | ignore if url-like |    never ignore   |
+    // |-------------------------------------------------------|
+    let ignore_urllike =
+        can_be_url || (ori_link.kind == LinkType::ShortcutUnknown && !ori_link.link.contains('`'));
+    if ignore_urllike && should_ignore_link(path_str) {
```

Comment:
It looks like the complete truth table is:

|              |  unknown shortcut link  | unknown reference / collapsed link |     other link     |
|--------------|-------------------------|------------------------------------|--------------------|
| has backtick |      never ignore       |            never ignore            | ignore if url-like |
| no backtick  |   ignore if url-like    |            never ignore            | ignore if url-like |

Assuming I'm understanding the logic correctly (the difference is to things like `[first](second)`).

Reply:
That is correct.  Technically I do explicitly define the scope of the existing table with the line "here's a truth table for how **link kinds that cannot be urls** are handled:"

the line is currently 66 chars long, i should be able to fit in the other case without upsetting tidy (100 char limit), do you want me to?